### PR TITLE
First stable version: Core API integration

### PR DESCRIPTION
**Pull request by teexone**
Malevich Meta is a user-friendly interface that enables you to communicate with Core API using simple, clean and concise Python scripts.

Includes functionality:
- Malevich CLI interface
- Package installation: `malevich use [IMAGE] [PACKAGE_NAME] [IMAGE_USER] [IMAGE_TOKEN]`
- `autoflow' engine: automatically reconstructing an execution tree by tracing arguments in functions
- Interpretation using Malevich Core API

### CODE REVIEW

File: README.md
Comment (zakhardavydov) [inline README.md:1]: We should add a end developer doc. The doc should cover:
- How to
- Key commands and expected outputs
- Guide on how to write a new metascript file

File: examples/duolingo.py
Comment (zakhardavydov) [inline examples/duolingo.py:1]: That looks like old example. Remove it and reference a separate duolingo repo

File: examples/sandbox.py
Comment (zakhardavydov) [inline examples/sandbox.py:1]: This is good, maybe add more comments on what happens on each line

File: malevich/__init__.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Collection: an object that is subject to be passed to processors
+    """
+    assert any([file is not None, data is not None]), "Either file or data must be provided"
+    if file:
+        assert data is None, "Cannot provide both file and data"
+        data = pd.read_csv(file)
+
+    name = name or uuid4().hex
+
+    from malevich.models.collection import Collection
```
Comment (zakhardavydov) [inline malevich/__init__.py:37]: Why this can't be a global import

File: malevich/__init__.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    return gn.tracer((name, {
+            '__collection__': Collection(collection_id=name, collection_data=data),
+        }
+    ))
+
+
+
+def pipeline(
+    interpreter: Optional[str] = None,
+    core_host: Optional[str] = 'https://core.onjulius.co',
```
Comment (zakhardavydov) [inline malevich/__init__.py:47]: Make the constant a global variable

Comment (zakhardavydov) [reply malevich/__init__.py:47]: In a separate constants.py file

File: malevich/__init__.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+) -> Callable[[], str]:
+
+    def wrapper(function: Callable[[], None]) -> Callable[[], str]:
+        from malevich._autoflow.manager import Flow
```
Comment (zakhardavydov) [inline malevich/__init__.py:51]: Again, why can't this be global

File: malevich/__init__.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        def fn(*args, **kwargs):  # noqa: ANN202, ANN002, ANN003
+            match (interpreter):
+                case 'core':
+                    from malevich.interpreter.viacore import CoreInterpreter
```
Comment (zakhardavydov) [inline malevich/__init__.py:55]: This too

File: malevich/_autoflow/tree.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+T = TypeVar("T")
+
+class ExecutionTree(Generic[T]):
+    tree: list[tuple[T, T]] = []
+
+
+    def put_edge(self, caller: T, callee: T, link: Any = None) -> None:
+        self.tree.append((caller, callee, link))
+
+    def traverse(self) -> Iterator[tuple[T, T, Any]]:
```
Comment (zakhardavydov) [inline malevich/_autoflow/tree.py:13]: 🔥

File: malevich/commands/use.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import typer
+
+from malevich.install.image import ImageInstaller
+from malevich.manifest import ManifestManager
+
+use = typer.Typer()
+
+@use.command('image')
+def install_from_image(
```
Comment (zakhardavydov) [inline malevich/commands/use.py:9]: Are we sure we don't need a Typer help text?

File: malevich/interpreter/viacore.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+
+class CoreInterpreter(Interpreter):
+    """Inteprets the flow using Malevich Core API"""
+
+    def interpret(
+        self,
+        tree: ExecutionTree[tracer],
+        core_auth: str = None,
+        core_host: str = "https://core.onjulius.co",
```
Comment (zakhardavydov) [inline malevich/interpreter/viacore.py:22]: Constant out

**Review by zakhardavydov — APPROVED** [review]

File: README.md
Comment (teexone) [reply README.md:1]: Are you talking about general README.md, or special README.dev.md? Do we differentiate between them and how?

Comment (zakhardavydov) [reply README.md:1]: lgtm

File: malevich/__init__.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+def flow(
+    interpreter: Optional[str] = None,
+    core_host: Optional[str] = DEFAULT_CORE_HOST
+) -> Callable[[], str]:
+
+    def wrapper(function: Callable[[], None]) -> Callable[[], str]:
+        def fn(*args, **kwargs):  # noqa: ANN202, ANN002, ANN003
+            match (interpreter):
+                case 'core':
```
Comment (zakhardavydov) [inline malevich/__init__.py:57]: Can we build an enum here?

```
from enum import Enum

class Platform(Enum):
  CORE = "core"
  SPACE = "space"
  
 
 Core = Platform.CORE
 Space = Platform.Space
```

File: malevich/interpreter/viacore.py
Comment (zakhardavydov) [inline malevich/interpreter/viacore.py:1]: As discussed on the sync, please take tree traversal logic into BaseInterpreter class and inherit from implementing app, collection, flow, link creation interfaces

**Review by zakhardavydov — CHANGES_REQUESTED** [review]
@teexone great stuff, please refactor interpreter before merging into stable

**Review by zakhardavydov — APPROVED** [review]
@teexone this is great. Feel free to merge!

**Pull request merged**